src/components/db-explorer.tsx

```diff
@@ -84,15 +85,10 @@ function DBExplorerInner(props: DBExplorerInnerProps) {
       connectionRef.current = c;
 
       try {
-        if (extension === "csv") {
-          await db.registerFileText(`data.csv`, content);
-          await c.insertCSVFromPath(`data.csv`, { name: "data" });
-        } else if (extension === "json") {
-          await db.registerFileText(`data.json`, content);
-          await c.insertJSONFromPath("data.json", { name: "data" });
-        }
+        await db.registerFileText(filename, content);
+        await c.insertCSVFromPath(filename, { name: filenameWithoutExtension });
         setDbStatus("success");
-        setQuery("select * from data");
+        setQuery(`select * from '${filenameWithoutExtension}'`);
```

Comment:
Minor point - there are repos where a dataset may be produced in two separate file formats, ie CSV and JSON. Not a problem now, but when (if!) flat-viewer is extended to have visibility over multiple files at any given time, the name collision may not be desirable, particularly if it turns out a repo owner deliberately persists _different_ data in eg dataset.csv and dataset.json.

That said, I concede that it is awkward having a file extension in a table name =)

Reply:
Good call out. I think for the time being we're OK here until `flat-viewer` does the multi-file visibility that you describe.